### PR TITLE
cppcheck performance fixes in utils/

### PR DESCRIPTION
**Pull request by tobbi**
Should be straightforward.

### CODE REVIEW

File: xbmc/utils/CryptThreading.cpp [hunk 1]
```diff
@@ -66,14 +66,14 @@ static unsigned long thread_id()
...
-  bool attemptedToSetSSLMTHook = false;
 #ifdef HAVE_OPENSSL
   // set up OpenSSL
   numlocks = CRYPTO_num_locks();
   CRYPTO_set_id_callback(thread_id);
   CRYPTO_set_locking_callback(lock_callback);
-  attemptedToSetSSLMTHook = true;
+  bool attemptedToSetSSLMTHook = true;
 #else
+  bool attemptedToSetSSLMTHook = false;
```
Comment (Jalle19) [inline xbmc/utils/CryptThreading.cpp]: Didn't this kind of change lead to a build error in one of your earlier PRs?

Comment (fritsch) [inline xbmc/utils/CryptThreading.cpp]: Nope - was a similar but not the same situation. In the old code it was defined in a real else { } clause and went out of scope directly after it.

But I would just not do such optimizations. It does not harm if it's set to false out of that block and set to true in the #ifdef block

Comment (mkortstiege) [inline xbmc/utils/CryptThreading.cpp]: I agree with @fritsch. @tobbi, what exactly is cppcheck reporting?

Comment (tobbi) [inline xbmc/utils/CryptThreading.cpp]: [CryptThreading.cpp:69] -> [CryptThreading.cpp:75]: (performance) Variable 'attemptedToSetSSLMTHook' is reassigned a value before the old one has been used.

Comment (Jalle19) [inline xbmc/utils/CryptThreading.cpp]: I'd consider that a false positive.

**Comment by tobbi** [comment]
I have excluded the CryptThreading.cpp changes from this PR.

**Pull request merged**